### DISCUSSION TITLE
How to work with an image blob, manipulate the image and show it in the browser?

### DISCUSSION THREAD

**Discussion by raphapaulino** [spatie/image#189] [Q&A]
Hi guys. 

First I want to congrats the spatie team for delivering us this awesome laravel package, thank you so much!

My challenge here is render an image after its manipulation like, for example, crop it. I have studied how to do this with **imagick** extension for php, than I have had this result:

```
<?php

ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

$url = 'https://spatie.be/docs/image/v1/images/example.jpg';

$image = file_get_contents($url);

$img = new Imagick();

$img->readImageBlob($image);

$img->cropThumbnailImage(120, 256);

ob_start();
print $img->getImageBlob(); 
$contents =  ob_get_contents();
ob_end_clean();
$img->destroy();

echo "<img src='data:image/jpg;base64,".base64_encode($contents)."' />";
```

How can I use this same approach with "spatie/image" package without necessarily saving the image on localhost? I didn't find the answer in the package documentation.

**Comment by luiseduardobraschi** [comment]
I don't  believe it's possible, but it would be a great addition to a future version.
